test/job.js

```diff
@@ -615,22 +615,21 @@ describe('Job', () => {
     });
 
     it('clears locks on stop', async() => {
-      agenda.define('longRunningJob', job => { // eslint-disable-line no-unused-vars
-        // Job never finishes
-      });
-      agenda.every('10 seconds', 'longRunningJob');
-      agenda.processEvery('1 second');
-
-      await agenda.start();
-      await delay(jobTimeout);
-      await agenda.stop();
+      try {
+        agenda.define('longRunningJob', job => { // eslint-disable-line no-unused-vars
+          // Job never finishes
+        });
+        agenda.every('10 seconds', 'longRunningJob');
+        agenda.processEvery('1 second');
 
-      agenda._collection.findOne({name: 'longRunningJob'}, (err, job) => {
```

Comment:
I suspect there was an await missing here, that caused the afterEach mocha section of the test to be executed clearing the jobs which eventually made the findOne return undefined, no job found.